memcached-operator/pkg/controller/memcached/memcached_controller.go

```diff
@@ -122,15 +132,31 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 
 	// Ensure the deployment size is the same as the spec
 	size := memcached.Spec.Size
-	if *found.Spec.Replicas != size {
-		found.Spec.Replicas = &size
-		err = r.client.Update(context.TODO(), found)
+	if *deployment.Spec.Replicas != size {
+		deployment.Spec.Replicas = &size
+		err = r.client.Update(context.TODO(), deployment)
 		if err != nil {
-			reqLogger.Error(err, "Failed to update Deployment.", "Deployment.Namespace", found.Namespace, "Deployment.Name", found.Name)
+			reqLogger.Error(err, "Failed to update Deployment.", "Deployment.Namespace", deployment.Namespace, "Deployment.Name", deployment.Name)
 			return reconcile.Result{}, err
 		}
-		// Spec updated - return and requeue
-		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if the service already exists, if not create a new one
+	// NOTE: The Service is used to expose the deployment. However, the service is not required at all for the memcached example to work. The purpose is to add more examples of what you can do in your operator project.
+	service := &corev1.Service{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: memcached.Name, Namespace: memcached.Namespace}, service)
+	if err != nil && errors.IsNotFound(err) {
+		// Define a new service
```

Comment:
These are nits, but just because this is an example we should be consistent with the comments.
```suggestion
		// Define a new Service object
```

Reply:
👍 done tks